[2024-10-09T11:19:37.803+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-09T11:19:37.852+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: mssql_test2.create_table_mssql_task manual__2024-10-09T11:19:34.663881+00:00 [queued]>
[2024-10-09T11:19:37.884+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: mssql_test2.create_table_mssql_task manual__2024-10-09T11:19:34.663881+00:00 [queued]>
[2024-10-09T11:19:37.885+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-10-09T11:19:37.917+0000] {taskinstance.py:2888} INFO - Executing <Task(MsSqlOperator): create_table_mssql_task> on 2024-10-09 11:19:34.663881+00:00
[2024-10-09T11:19:37.939+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=340) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-10-09T11:19:37.951+0000] {standard_task_runner.py:72} INFO - Started process 342 to run task
[2024-10-09T11:19:37.951+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'mssql_test2', 'create_table_mssql_task', 'manual__2024-10-09T11:19:34.663881+00:00', '--job-id', '45', '--raw', '--subdir', 'DAGS_FOLDER/mssql_test2.py', '--cfg-path', '/tmp/tmp1dbw97uj']
[2024-10-09T11:19:37.957+0000] {standard_task_runner.py:105} INFO - Job 45: Subtask create_table_mssql_task
[2024-10-09T11:19:38.080+0000] {task_command.py:467} INFO - Running <TaskInstance: mssql_test2.create_table_mssql_task manual__2024-10-09T11:19:34.663881+00:00 [running]> on host 37a08abe3d9a
[2024-10-09T11:19:38.452+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='mssql_test2' AIRFLOW_CTX_TASK_ID='create_table_mssql_task' AIRFLOW_CTX_EXECUTION_DATE='2024-10-09T11:19:34.663881+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-09T11:19:34.663881+00:00'
[2024-10-09T11:19:38.457+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2024-10-09T11:19:38.457+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2024-10-09T11:19:38.458+0000] {logging_mixin.py:190} INFO - Current task name:create_table_mssql_task state:running start_date:2024-10-09 11:19:37.853843+00:00
[2024-10-09T11:19:38.458+0000] {logging_mixin.py:190} INFO - Dag name:mssql_test2 and current dag run status:running
[2024-10-09T11:19:38.458+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-09T11:19:38.459+0000] {sql.py:266} INFO - Executing: 
    USE master;
    CREATE TABLE IF NOT EXISTS Country (
        name VARCHAR(255) NOT NULL,
        continent VARCHAR(255) NOT NULL
    );
    
[2024-10-09T11:19:38.478+0000] {base.py:84} INFO - Retrieving connection '***_mssql'
[2024-10-09T11:19:38.491+0000] {base.py:84} INFO - Retrieving connection '***_mssql'
[2024-10-09T11:19:38.528+0000] {sql.py:542} INFO - Running statement: 
    USE master;
    CREATE TABLE IF NOT EXISTS Country (
        name VARCHAR(255) NOT NULL,
        continent VARCHAR(255) NOT NULL
    );
    , parameters: None
[2024-10-09T11:19:38.530+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "src/pymssql/_pymssql.pyx", line 447, in pymssql._pymssql.Cursor.execute
  File "src/pymssql/_mssql.pyx", line 1125, in pymssql._mssql.MSSQLConnection.execute_query
  File "src/pymssql/_mssql.pyx", line 1156, in pymssql._mssql.MSSQLConnection.execute_query
  File "src/pymssql/_mssql.pyx", line 1289, in pymssql._mssql.MSSQLConnection.format_and_run_query
  File "src/pymssql/_mssql.pyx", line 1855, in pymssql._mssql.check_cancel_and_raise
  File "src/pymssql/_mssql.pyx", line 1901, in pymssql._mssql.raise_MSSQLDatabaseException
pymssql._mssql.MSSQLDatabaseException: (156, b"Incorrect syntax near the keyword 'IF'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/common/sql/operators/sql.py", line 272, in execute
    output = hook.run(
             ^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/common/sql/hooks/sql.py", line 487, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/common/sql/hooks/sql.py", line 547, in _run_command
    cur.execute(sql_statement)
  File "src/pymssql/_pymssql.pyx", line 465, in pymssql._pymssql.Cursor.execute
pymssql.exceptions.OperationalError: (156, b"Incorrect syntax near the keyword 'IF'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n")
[2024-10-09T11:19:38.547+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2024-10-09T11:19:38.547+0000] {logging_mixin.py:190} INFO - Task start:2024-10-09 11:19:37.853843+00:00 end:2024-10-09 11:19:38.546197+00:00 duration:0.692354
[2024-10-09T11:19:38.548+0000] {logging_mixin.py:190} INFO - Task:<Task(MsSqlOperator): create_table_mssql_task> dag:<DAG: mssql_test2> dagrun:<DagRun mssql_test2 @ 2024-10-09 11:19:34.663881+00:00: manual__2024-10-09T11:19:34.663881+00:00, state:running, queued_at: 2024-10-09 11:19:34.690502+00:00. externally triggered: True>
[2024-10-09T11:19:38.548+0000] {logging_mixin.py:190} INFO - Failure caused by (156, b"Incorrect syntax near the keyword 'IF'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n")
[2024-10-09T11:19:38.548+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=mssql_test2, task_id=create_table_mssql_task, run_id=manual__2024-10-09T11:19:34.663881+00:00, execution_date=20241009T111934, start_date=20241009T111937, end_date=20241009T111938
[2024-10-09T11:19:38.570+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-09T11:19:38.571+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 45 for task create_table_mssql_task ((156, b"Incorrect syntax near the keyword 'IF'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n"); 342)
Traceback (most recent call last):
  File "src/pymssql/_pymssql.pyx", line 447, in pymssql._pymssql.Cursor.execute
  File "src/pymssql/_mssql.pyx", line 1125, in pymssql._mssql.MSSQLConnection.execute_query
  File "src/pymssql/_mssql.pyx", line 1156, in pymssql._mssql.MSSQLConnection.execute_query
  File "src/pymssql/_mssql.pyx", line 1289, in pymssql._mssql.MSSQLConnection.format_and_run_query
  File "src/pymssql/_mssql.pyx", line 1855, in pymssql._mssql.check_cancel_and_raise
  File "src/pymssql/_mssql.pyx", line 1901, in pymssql._mssql.raise_MSSQLDatabaseException
pymssql._mssql.MSSQLDatabaseException: (156, b"Incorrect syntax near the keyword 'IF'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/common/sql/operators/sql.py", line 272, in execute
    output = hook.run(
             ^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/common/sql/hooks/sql.py", line 487, in run
    self._run_command(cur, sql_statement, parameters)
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/common/sql/hooks/sql.py", line 547, in _run_command
    cur.execute(sql_statement)
  File "src/pymssql/_pymssql.pyx", line 465, in pymssql._pymssql.Cursor.execute
pymssql.exceptions.OperationalError: (156, b"Incorrect syntax near the keyword 'IF'.DB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\nDB-Lib error message 20018, severity 15:\nGeneral SQL Server error: Check messages from the SQL Server\n")
[2024-10-09T11:19:38.626+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-09T11:19:38.655+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/models/baseoperator.py:1378: AirflowProviderDeprecationWarning: Call to deprecated class MsSqlOperator. (Please use `***.providers.common.sql.operators.sql.SQLExecuteQueryOperator`.Also, you can provide `hook_params={'schema': <database>}`.)
  result = cls.__new__(cls)

[2024-10-09T11:19:38.723+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-09T11:19:38.734+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
